packages/server/src/migrations/0.1.21-client-tokens.ts

```diff
@@ -0,0 +1,47 @@
+import { Migration } from '@botpress/messaging-engine'
+import { v4 as uuidv4 } from 'uuid'
+
+export class ClientTokensMigration extends Migration {
+  meta = {
+    name: ClientTokensMigration.name,
+    description: 'Removes the token field from the msg_clients table',
+    version: '0.1.21'
+  }
+
+  async valid() {
+    return (await this.trx.schema.hasTable('msg_clients')) && this.trx.schema.hasTable('msg_client_tokens')
+  }
+
+  async applied() {
+    return !(await this.trx.schema.hasColumn('msg_clients', 'token'))
+  }
+
+  async up() {
+    const clients = await this.trx('msg_clients')
+
+    for (const client of clients) {
+      await this.trx('msg_client_tokens').insert({ id: uuidv4(), clientId: client.id, token: client.token })
+    }
```

Comment:
It would be possible to replace the uuidv4() here by a deterministic algorithm that takes the token and creates a uuid. This way it would be possible to upgrade / downgrade the token from anywhere, without having to look at the db or adding an api point

Reply:
Actually nevermind there's no way for the other end to know what hash was produced, since it's salted